app/utils/results.py

```diff
@@ -223,3 +223,90 @@ def add_ip_card(html_soup: BeautifulSoup, ip: str) -> BeautifulSoup:
         # Inserting the element
         ref_element.insert_before(ip_tag)
     return html_soup
+
+
+def check_currency(response: str):
+    """Check whether the results have currency conversion
+
+    Args:
+        response: Search query Result
+
+    Returns:
+        list
+
+    """
+    soup = BeautifulSoup(response, 'html.parser')
+    a = soup.find(class_='ZINbbc xpd O9g5cc uUPGi')
+    tag = a.find('a')
+    if tag and tag['href'] == 'https://g.co/gfd':
+        currency1 = soup.find(class_="xUrNXd UMOHqf")
+        currency2 = soup.select_one('.kCrYT'
+                                    ' .BNeawe.iBp4i.AP7Wnd '
+                                    '.BNeawe.iBp4i.AP7Wnd')
+        currency1 = currency1.text.rstrip('=').split(' ', 1)
+        currency2 = currency2.text.split(' ', 1)
+        return [float(currency1[0]), currency1[1],
+                float(currency2[0]), currency2[1]]
+    return None
+
+
+def add_currency_b(soup: BeautifulSoup,
+                   conversion_details: list) -> BeautifulSoup:
+    """Adds the currency conversion boxes
+    to response of the search query
+
+    Args:
+        soup: Parsed search result
+        conversion_details: list of currency
+        related information
+
+    Returns:
+        BeautifulSoup
+    """
+    # Element before which the code will be changed
+    # (This is the 'disclaimer' link)
+    element1 = soup.select_one('[class="nXE3Ob"]')
+
+    # Creating the conversion factor
+    conversion_factor = conversion_details[0]/conversion_details[2]
+
+    # Creating a new div for the input boxes
+    conversion_box = soup.new_tag('div')
+    conversion_box['class'] = 'conversion_box'
+
+    # Currency to be converted from
+    input_box1 = soup.new_tag('input')
+    input_box1['id'] = 'cb1'
+    input_box1['type'] = 'number'
+    input_box1['class'] = 'cb Hg3mWc'
```

Comment:
Do we have to call this class `Hg3mWc`? Or can we just extract the styles from that class, rename it something clearer (`currency_element` or something, idk) and include the styling in the main css?

Reply:
I will rename it. I just had it to keep it consistent with the search results but keeping it clearer would be better yes.